### PR TITLE
Support proto3-suite 0.5.0 and later.

### PR DESCRIPTION
**Pull request by j6carey**
Starting with proto3-suite 0.5.0, the initial character of
a service method names is no longer forced to uppercase.

### CODE REVIEW

**Review by jcarr-awake — APPROVED** [review]
This is clear enough for me, my I-dont-know-anything-about-the-original-change question is why we wouldn't be able to just use prefixedFieldName always, unless it didn't exist?

**Comment by j6carey** [comment]
@Friede80, @jcarr-awake ,

`prefixedMethodName` was added by https://github.com/awakesecurity/proto3-suite/pull/171 (first appearing in proto3-suite 0.5.0) and used for service method names only, with `prefixedFieldName` still used for other things.  Apparently the desire is to make it possible for two methods of the same service to differ only in the case of the first letter of their method names.

I ran into a compilation failure in an Awake application of this library before making this change, because the Haskell spelling of at least one service method was wrong.

Regarding other uses of `prefixedFieldName` in `grpc-mqtt`: thanks for spotting that problem with this PR; I'll push up a change.

**Comment by j6carey** [comment]
c346d5762b1cf8a3463a012eb77fcd1c98289b6a replaces the other uses of `prefixedFieldName` and isolates the conditional code to a compatibility module.

File: src/Proto3/Suite/DotProto/Internal/Compat.hs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+-- an alias of `Proto3.Suite.DotProto.Internal.prefixedFieldName`.
+module Proto3.Suite.DotProto.Internal.Compat (prefixedMethodName) where
+
+#if MIN_VERSION_proto3_suite(0,5,0)
+
+import Proto3.Suite.DotProto.Internal (prefixedMethodName)
+
+#else
+
+import Proto3.Suite.DotProto.Internal (prefixedFieldName)
```
Comment (Friede80) [inline src/Proto3/Suite/DotProto/Internal/Compat.hs:18]: Need these extra imports in this case.
```suggestion
import Control.Monad.Except (MonadError)
import Proto3.Suite.DotProto.Internal (CompileError, prefixedFieldName)
import Relude (String)
```

**Review by Friede80 — COMMENTED** [review]
This looks good. Thanks for moving it out to a separate module.

It does raise a separate issue that the tests fail when using `proto3-suite` >= 0.5.0, because the generated code is using version 0.4.3. I think we are still good to merge this PR and I'll create an issue for fixing the tests.

**Review by Friede80 — APPROVED** [review]

**Comment by j6carey** [comment]
@Friede80 , I think the test failures were a legitimate concern.  I see two solutions: use the old name creation function for the functions that grpc-mqtt adds that are not in the .proto file, or just capitalize their names in the grpc-mqtt source so that both old and new versions of proto3-suite will produce the same compound name.  I adopted the latter solution in ab569abb53db78f2a4c724bed6b29ba1eeacaef3 .

**Pull request merged**